light-client/src/supervisor.rs

```diff
@@ -17,21 +17,26 @@ use tendermint::evidence::{ConflictingHeadersEvidence, Evidence};
 use contracts::pre;
 use crossbeam_channel as channel;
 
-/// Type alias for readability
+/// Verification operation [`Result`] type alias for readability.
 pub type VerificationResult = Result<LightBlock, Error>;
 
-/// Events which are exchanged between the `Supervisor` and its `Handle`s.
+/// Input events which are sent by the [`Handle`] to its [`Supervisor`]..
 #[derive(Debug)]
-pub enum Event {
-    // Inputs
+pub enum Input {
```

Comment:
Might be a good call to split this up. 
Throwing in the suggestion to rename this to `InputEvents` / `OutputEvents`?